internal/pkg/name-conversions/conversions.go

```diff
@@ -0,0 +1,90 @@
+package nameconversions
+
+import (
+	"fmt"
+
+	"github.com/confluentinc/cli/internal/pkg/errors"
+)
+
+const (
+	ResourceNameNotFoundErrorMsg          = `resource with name or ID "%s" was not found`
+	DuplicateResourceNameErrorMsg         = `the resource name "%s" is shared across multiple resources`
+	DuplicateResourceNameErrorSuggestions = "retry the previous command using a resource id"
+)
+
+type resource any
+
+type resourcePtr interface {
+	GetId() string
+}
+
+type specPtr interface {
+	GetDisplayName() string
+}
+
+type v2ResourcePtr interface {
```

Comment:
Can you document the difference between `resourcePtr` and `v2ResourcePtr`? Not sure I understand.